### ISSUE TITLE
MOF compiler embedded instance. EOL apparently illegal character

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbem#2330]
Update by Andy: Edited formatting, removed duplicate sentence in MOF.

When compiling a an instance with an embeddedinstance property (i.e. contains a string with the MOF for the embedded instance). The compile fails if the string contains the \n character.

```
instance of TST_Foo {
   InstanceID = "good_inst_mof";
   Embedded1 =
      "instance of TST_Embedded {\n   FQLSupported = true;\n   Validated = "
      "true;\n};\n";
};
```

produces:

Comment by Andy: I assume the following should have been a repr of the resulting CIMInstance object?

```
instance of TST_Foo {
   InstanceID = "good_inst_mof";
   Embedded1 =
      "instance of TST_Embedded {\n   FQLSupported = true;\n   Validated = "
      "true;\n};\n";
};
```

However, add a single EOL to the string as follows:

```
instance_mof = """
instance of TST_Foo {
    InstanceID = "good_inst_mof";
    Embedded1 =
        "instance of TST_Embedded {\n"
        "FQLSupported = true;"
        "Validated = true;"
        "};";
};
```

and the compile fails:

```
   Traceback (most recent call last):
  File "./test_inst.py", line 83, in <module>
    conn.compile_mof_string(instance_mof)
  File "/home/kschopmeyer/pywbem/githubpywbem/pywbem/pywbem_mock/_wbemconnection_mock.py", line 602, in compile_mof_string
    mofcomp.compile_string(mof_str, namespace)
  File "/home/kschopmeyer/pywbem/githubpywbem/pywbem/pywbem/_mof_compiler.py", line 2819, in compile_string
    rv = self.parser.parse(mof, lexer=lexer)
  File "/home/kschopmeyer/virtualenvs/pywbem3/lib/python3.6/site-packages/ply/yacc.py", line 333, in parse
    return self.parseopt_notrack(input, lexer, debug, tracking, tokenfunc)
  File "/home/kschopmeyer/virtualenvs/pywbem3/lib/python3.6/site-packages/ply/yacc.py", line 1201, in parseopt_notrack
    tok = call_errorfunc(self.errorfunc, errtoken, self)
  File "/home/kschopmeyer/virtualenvs/pywbem3/lib/python3.6/site-packages/ply/yacc.py", line 192, in call_errorfunc
    r = errorfunc(token)
  File "/home/kschopmeyer/pywbem/githubpywbem/pywbem/pywbem/_mof_compiler.py", line 549, in p_error
    raise MOFParseError(msg=msg, parser_token=p)
pywbem._mof_compiler.MOFParseError: MOF parsing error:String:5:12 Illegal character '"'
    instance of TST_Foo {
        InstanceID = "good_inst_mof";
        Embedded1 =
            "instance of TST_Embedded {
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

NOTE: The classes for this compile are:

```
    Qualifier Key : boolean = false,
        Scope(property, reference),
        Flavor(DisableOverride, ToSubclass);

    Qualifier Description : string = null,
        Scope(any),
        Flavor(EnableOverride, ToSubclass, Translatable);

    Qualifier EmbeddedInstance : string = null,
        Scope(property, method, parameter);

    Qualifier EmbeddedObject : boolean = false,
        Scope(property, method, parameter),
        Flavor(DisableOverride, ToSubclass);

    class TST_Embedded {
        boolean FQLSupported;
        boolean CQLSupport;
        boolean Validated;
    };

         [Description ("blah blah")]
    class TST_Foo {
            [Key, Description ("This is key prop")]
        string InstanceID;
            [Description ( "Allocation state of the managed element." ),
                EmbeddedInstance ( "TST_Embedded" )]
        string Embedded1;
    };
```

**Comment by andy-maier** [comment]
I suggest that you create a PR that adds this testcase that fails and demonstrates the error.

**Comment by andy-maier** [comment]
We agreed top move this one out to 1.0.0 final

**Comment by andy-maier** [comment]
I think it boil down to these two things:
1. The "\n" in a Python string is interpreted by Python and turned into a real newline character, which makes the closing double quote of the MOF string part to be on a new line. Using double escapes or "r" in front of the string should address that.
2. The rules in DSP0004 on how mandatory the use of string ecape sequences is (vs. using the actual character) are to be discussed. I believe that our MOF compiler assumes these escape rules must be used. At least, for the newline it seems to do that. So clarifying that in DMTF is another thing to do.

**Comment by andy-maier** [comment]
Fixed in PR #2350
